File: client/src/app/shared/models/character-spell-slots.ts
export interface CharacterSpellSlots {
  id: number;

  // Maximum spell slots per level
  maxLevel1: number;
  maxLevel2: number;
  maxLevel3: number;
  maxLevel4: number;
  maxLevel5: number;
  maxLevel6: number;
  maxLevel7: number;
  maxLevel8: number;
  maxLevel9: number;

  // Used spell slots per level
  usedLevel1: number;
  usedLevel2: number;
  usedLevel3: number;
  usedLevel4: number;
  usedLevel5: number;
  usedLevel6: number;
  usedLevel7: number;
  usedLevel8: number;
  usedLevel9: number;
}

File: Core/Entities/CharacterSpellSlots.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.Entities;

public class CharacterSpellSlots : BaseEntity
{
    public int MaxLevel1 { get; set; }
    public int MaxLevel2 { get; set; }
    public int MaxLevel3 { get; set; }
    public int MaxLevel4 { get; set; }
    public int MaxLevel5 { get; set; }
    public int MaxLevel6 { get; set; }
    public int MaxLevel7 { get; set; }
    public int MaxLevel8 { get; set; }
    public int MaxLevel9 { get; set; }

    public int UsedLevel1 { get; set; }
    public int UsedLevel2 { get; set; }
    public int UsedLevel3 { get; set; }
    public int UsedLevel4 { get; set; }
    public int UsedLevel5 { get; set; }
    public int UsedLevel6 { get; set; }
    public int UsedLevel7 { get; set; }
    public int UsedLevel8 { get; set; }
    public int UsedLevel9 { get; set; }

    public int CharacterId { get; set; }
}
